### PR TITLE
Setup docs: Add installation instructions for Debian and Fedora OS #8491

### LINKED ISSUE #8491: Setup docs: Add installation instructions for Debian and Fedora OS
In documentation for [setting up environment](https://github.com/TEAMMATES/teammates/blob/master/docs/setting-up.md) , the Java API plugin for Google cloud requires installation as separate packages in Linux system before it can be used by `gcloud` module. The current documentation does not cover this up.

### PR DESCRIPTION
**Pull request by Progyan1997**
Sepearte package available for Google cloud Java API plugin. For Linux system, the package can be directly installed.

Signed-off-by: Progyan Bhattacharya <progyanb@acm.org>

Fixes #8491 

**PR Checklist**

Ensure that you have:
- [ ] Read and understood our PR guideline: https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-4-submit-a-pr
  - [ ] Added the issue number to the "Fixes" keyword above
  - [ ] Titled the PR as specified in the abovementioned document
- [ ] Made your changes on a branch other than `master` and `release`
- [ ] Gone through all the changes in this PR and ensured that:
  - [ ] They addressed one (and only one) issue
  - [ ] No unintended changes were made
- [ ] Run and passed static analysis: `./gradlew lint` and `npm run lint`
- [ ] Added/updated tests, if changes in functionality were involved
- [ ] Added/updated documentation to public APIs (classes, methods, variables), if applicable

**Outline of Solution**

### CODE REVIEW

**Comment by tshradheya** [comment]
@Progyan1997 TEAMMATES follows the following [process](https://github.com/TEAMMATES/teammates/blob/master/docs/process.md) for contribution. 

So before you open a PR, it would be necessary to open an issue about what you aim to do via the PR and get the approval from the project maintainers that the issue is valid and needs to be worked upon.

**Comment by tshradheya** [comment]
Yes, please open an issue. It doesn't matter that even if it is just for Documentation

**Comment by wkurniawan07** [comment]
Hi @Progyan1997, thanks for the contribution.
However, this does not look like something we really need because the current steps as detailed in the setup document works also for Linux machines, and we don't want to be duplicates of the official setup guide from the Cloud SDK themselves (otherwise we also have to document the `apt-get` alternative and all the what-nots, right?).

**Comment by Progyan1997** [comment]
@wkurniawan07 I am already working on `apt` for Debian/Ubuntu systems. And the given setup no longer works with RHEL-based systems (at least)

**Comment by wkurniawan07** [comment]
@Progyan1997 Are you sure just downloading the Java SDK works?

**Comment by Progyan1997** [comment]
@wkurniawan07 Yes, that's the way I followed to work with!

**Comment by wkurniawan07** [comment]
@Progyan1997 as in you don't need to install the Cloud SDK first?

**Comment by Progyan1997** [comment]
@wkurniawan07 You must install `gcloud` package first, although if you don't it will install it anyway for the dependencies. But my point was `gcloud` is not allowed to install `java-plugin`, it must be installed separately as a package.

**Comment by wkurniawan07** [comment]
@Progyan1997 are you sure the install commands with `apt-get` or `yum` work just like that without setting up the configurations?

**Comment by Progyan1997** [comment]
@wkurniawan07 Please explain what you said.

**Comment by wkurniawan07** [comment]
According to [this](https://cloud.google.com/sdk/docs/quickstart-debian-ubuntu) and [this](https://cloud.google.com/sdk/docs/quickstart-redhat-centos) (the "Before you begin" part) there are some things that need to be done.

**Comment by Progyan1997** [comment]
@wkurniawan07 That is for installing `gcloud`, the parent module. I think it's already present in the documentation.

**Comment by wkurniawan07** [comment]
@Progyan1997 it is not; our current setup docs make no mention to `yum`/`apt-get`-based installation other than a reference to the official installation page. So, installing `app-engine-java` with `apt-get` or `yum` works without the need for the configuration set out in the "Before you begin" part?

**Comment by Progyan1997** [comment]
@wkurniawan07 Then should I add those information as well?

**Comment by wkurniawan07** [comment]
Before you proceed further, have you tried that the entire setup (up to building/testing/deploying) works in any one of those OSes?
The setting up/development guideline is field-tested and actively maintained for Linux (Travis CI, although strictly speaking it's Ubuntu Trusty), Windows (AppVeyor), and Mac OS (my own setup). If we want to add support for any new OS, we need to make sure that (1) the entire routine works and (2) the setup is at least used by someone. We don't want to end up adding something now only for someone later to complain "this instruction does not work for me" and we don't have any resource/know-how to help.

**Comment by Progyan1997** [comment]
@wkurniawan07 
> Before you proceed further, have you tried that the entire setup (up to building/testing/deploying) works in any one of those OSes?

Yes, I did indeed. The given procedure did not work with CentOS 7, at least that was my experience. So I had to follow the official documentation and proceed with the mentioned steps. Since I am adding `yum` support, I thought of providing `apt` package informations as well.

> We don't want to end up adding something now only for someone later to complain "this instruction does not work for me" 

The previous documentations are kept as is, because most probably it works with other Linux distributions as well as Windows. Personally I would recommend to follow them, and then mine only if it does not work.

**Comment by wkurniawan07** [comment]
I'd recommend the following:
- In the step which installs Google Cloud SDK, add a reference to the official docs for Debian/Ubuntu/Red Hat/CentOS. Something like "If you use Debian/Ubuntu/Red Hat/CentOS, the installation is slightly more complicated. Refer to the official documentation provided by Google.". It is too wordy for everything to be duplicated here.
- In the step which installs App Engine Java, it is acceptable to mention `yum`/`apt-get` provided the above is done, but you don't need new code blocks for that (and the header is rather disruptive). Just do something like:
  ```sh
  # Linux/OS X/Windows
  gcloud -q components install app-engine-java

  # Debian/Ubuntu
  sudo apt-get install google-cloud-sdk-app-engine-java
  ```

**Comment by Progyan1997** [comment]
@wkurniawan07 
> In the step which installs Google Cloud SDK, add a reference to the official docs...
> ... you don't need new code blocks for that...

Made the above changes as you recommended.

**Comment by Progyan1997** [comment]
@wkurniawan07 Have a look.

File: docs/setting-up.md [hunk 1]
```diff
@@ -50,15 +50,31 @@ More information can be found at [this documentation](https://help.github.com/ar
...
    ./install.sh --path-update true
+
    # Windows
    install.bat --path-update true
+
+   # Red Hat/CentOS/Fedora
+   sudo yum install google-cloud-sdk
+
+   # Debian/Ubuntu
+   sudo apt-get install google-cloud-sdk
```
Comment (wkurniawan07) [inline docs/setting-up.md]: No need to include them for the following reasons:
- They are mentioned in the official docs
- If you only display them like this, it's unlikely to work because the "Before you begin" steps are not done

If anything, just do what I suggested: "If you use Debian/Ubuntu/Red Hat/CentOS, the installation is slightly more complicated. Refer to the official documentation provided by Google."

File: docs/setting-up.md [hunk 2]
```diff
@@ -50,15 +50,31 @@ More information can be found at [this documentation](https://help.github.com/ar
...
+   # Linux/OS X/Windows
    gcloud -q components install app-engine-java
+   
+   # Red Hat/CentOS/Fedora
+   sudo yum install google-cloud-sdk-app-engine-java
+   
+   # Debian/Ubuntu
+   sudo apt-get install google-cloud-sdk-app-engine-java
    ```
+   If you face any problem regarding the installation, refer to the official documentation of Google Cloud Engine for [Debian/Ubuntu](https://cloud.google.com/sdk/docs/quickstart-debian-ubuntu) or [RHEL/CentOS/Fedora](https://cloud.google.com/sdk/docs/quickstart-redhat-centos).
```
Comment (wkurniawan07) [inline docs/setting-up.md]: This is about installing the Cloud SDK; if everything was done right before, the `yum` and `apt-get` commands down here would have worked.

File: docs/setting-up.md (hunk 1 above)
Comment (Progyan1997) [reply docs/setting-up.md]: Okay :+1:

File: docs/setting-up.md [hunk 3]
```diff
@@ -50,14 +50,24 @@ More information can be found at [this documentation](https://help.github.com/ar
 
    # Linux/OS X
    ./install.sh --path-update true
+
    # Windows
    install.bat --path-update true
    ```
+   If you are installing in Red Hat/CentOS/Fedora/Debian/Ubuntu follow the official documentation of Google Cloud Engine for [Debian/Ubuntu](https://cloud.google.com/sdk/docs/quickstart-debian-ubuntu) or [RHEL/CentOS/Fedora](https://cloud.google.com/sdk/docs/quickstart-redhat-centos) respectively.
```
Comment (wkurniawan07) [inline docs/setting-up.md]: Minor readability tweaks:
- Red Hat/CentOS/Fedora/Debian/Ubuntu -> Red Hat, CentOS, Fedora, Debian, or Ubuntu
- Ubuntu follow -> Ubuntu, refer to
- official documentation -> quickstart guide
- Google Cloud Engine -> Google Cloud SDK
- RHEL -> Red Hat

Comment (Progyan1997) [reply docs/setting-up.md]: On it... :wink:
